src/action.c

```diff
@@ -1046,6 +1057,19 @@ actions_run(struct view *activator, struct server *server,
 				view_set_shade(view, false);
 			}
 			break;
+		case ACTION_TYPE_TOGGLE_MAGNIFY:
+			magnify_toggle();
+			break;
+		case ACTION_TYPE_SET_MAGNIFICATION:
+			{
+				const char *dir = action_get_str(action, "scale", NULL);
+				if (!strcmp(dir, "up")) {
```

Comment:
This will break when the argument is not set, please add a branch in `action_is_valid()`

Another option could be to use two actions: `ZoomIn` and `ZoomOut`.

Reply:
> Another option could be to use two actions: `ZoomIn` and `ZoomOut`.

Much better idea! I'll do that.